resources/views/livewire/project/application/general.blade.php

```diff
@@ -109,35 +109,61 @@ class="underline" href="https://coolify.io/docs/knowledge-base/docker/registry"
-                                placeholder="Empty means it won't push the image to a docker registry."
-                                label="Docker Image" />
-                            <x-forms.input id="application.docker_registry_image_tag"
-                                placeholder="Empty means only push commit sha tag."
-                                helper="If set, it will tag the built image with this tag too. <br><br>Example: If you set it to 'latest', it will push the image with the commit sha tag + with the latest tag."
-                                label="Docker Image Tag" />
+                            @if (
+                                $application->destination->server->isSwarm() ||
+                                    $application->additional_servers->count() > 0 ||
+                                    $application->settings->is_build_server_enabled)
+                                <x-forms.input id="application.docker_registry_image_name" required label="Docker Image"
+                                    placeholder="Required!" />
+                                <x-forms.input id="application.docker_registry_image_tag"
+                                    helper="If set, it will tag the built image with this tag too. <br><br>Example: If you set it to 'latest', it will push the image with the commit sha tag + with the latest tag."
+                                    placeholder="Empty means latest will be used." label="Docker Image Tag" />
+                            @else
+                                <x-forms.input id="application.docker_registry_image_name"
+                                    helper="Empty means it won't push the image to a docker registry."
+                                    placeholder="Empty means it won't push the image to a docker registry."
+                                    label="Docker Image" />
+                                <x-forms.input id="application.docker_registry_image_tag"
+                                    placeholder="Empty means only push commit sha tag."
+                                    helper="If set, it will tag the built image with this tag too. <br><br>Example: If you set it to 'latest', it will push the image with the commit sha tag + with the latest tag."
+                                    label="Docker Image Tag" />
+                            @endif
                         @endif
+                    </div>
+
+                    @if ($application->build_pack === 'dockerimage')
+                        <div class="pt-4 grid grid-cols-2 items-center gap-4">
+                            <div class="w-fit">
+                                <x-forms.checkbox wire:model.live="application.docker_use_custom_registry"
+                                    id="application.docker_use_custom_registry"
+                                    helper="Select a registry to pull the image from." label="Use Private Registry" />
```

Comment:
One more thing: A dropdown is a bit limited, if you have docker hub and ghrc private images it should probably be a multiselect so you can select multiple private registries and these will be used.

Reply:
> Please also add the private registry section to all git deploy types under docker image:

This might require a bit of time, not yet sure how other things are built/handled and how i would implement there, but probably doable. Its just this one was the most logical since its not connected to anything, its just plain docker image pulled from dockerhub, but now it can be pulled from wherever